PKGBUILD

```diff
@@ -27,13 +28,82 @@ md5sums=('a3206df1c1b97df7a4ddcdd17cb97d0c'
+    aarch64) makedepends+=('gcc-arm-none-eabi' 'gcc')	
+            ;;	
+    x86_64) source+=('gcc-arm-8.2-2019.01-x86_64-arm-eabi.tar.xz::https://developer.arm.com/-/media/Files/downloads/gnu-a/8.2-2019.01/gcc-arm-8.2-2019.01-x86_64-arm-eabi.tar.xz?revision=72b9b9ec-30c5-4543-ae95-3d840eb01dae&la=en&hash=C898D59AC42008ED0527538ADF200C11F3E8F53A'	
+             'gcc-arm-8.2-2019.01-x86_64-aarch64-linux-gnu.tar.xz::https://developer.arm.com/-/media/Files/downloads/gnu-a/8.2-2019.01/gcc-arm-8.2-2019.01-x86_64-aarch64-linux-gnu.tar.xz?revision=21270570-4ec0-4bad-a9e1-09707614066a&la=en&hash=AFEDF645AF5B94336DB4E1E608806CEC87A02B8A')	
+            md5sums+=('fc11c4f55085c613133d7527a00fed8a'	
+                      'ed467a18abc7cf81d53c0cf6014b1867')          	
+            _pkgarch="aarch64"	
+            CARCH="aarch64"	
+            CHOST="aarch64-linux-gnu-"	
+            export PATH="${srcdir}/aarch64/bin:${PATH}"	
+            export ARCH='aarch64'	
+            export CROSS_COMPILE='aarch64-linux-gnu-'	
+            ;;	
+esac	
+
 prepare() {
   cd ${srcdir}/${_srcname}
   patch -p1 -i ../add-board-orangepi-4.patch
+
+  if [ -e ${srcdir}/gcc-arm-8.2-2019.01-x86_64-aarch64-linux-gnu.tar.xz ]; then	
+    if [ -d ${srcdir}/aarch64 ]; then	
+        rm -r ${srcdir}/aarch64 	
+    fi	
+    echo "Preparing Cross-Compilers"	
+    cd ${srcdir}/	
+    mkdir aarch64	
+    cd aarch64	
+    cp -r ../gcc-arm-8.2-2019.01-x86_64-aarch64-linux-gnu/* ./	
+    cp -r ../gcc-arm-8.2-2019.01-x86_64-arm-eabi/* ./	
+    echo "symbolic linking arm-eabi to arm-none-eabi"	
+    cd bin	
+    ln -s arm-eabi-addr2line arm-none-eabi-addr2line	
+    ln -s arm-eabi-c++filt arm-none-eabi-c++filt	
+    ln -s arm-eabi-gcc arm-none-eabi-gcc	
+    ln -s arm-eabi-gcc-ranlib arm-none-eabi-gcc-ranlib	
+    ln -s arm-eabi-gdb arm-none-eabi-gdb	
+    ln -s arm-eabi-ld.bfd arm-none-eabi-ld.bfd	
+    ln -s arm-eabi-ranlib arm-none-eabi-ranlib	
+    ln -s arm-eabi-strip arm-none-eabi-strip	
+    ln -s arm-eabi-ar arm-none-eabi-ar	
+    ln -s arm-eabi-cpp arm-none-eabi-cpp	
+    ln -s arm-eabi-gcc-8.2.1 arm-none-eabi-gcc-8.2.1	
+    ln -s arm-eabi-gcov arm-none-eabi-gcov	
+    ln -s arm-eabi-gfortran arm-none-eabi-gfortran	
+    ln -s arm-eabi-nm arm-none-eabi-nm	
+    ln -s arm-eabi-readelf arm-none-eabi-readelf	
+    ln -s arm-eabi-as arm-none-eabi-as	
+    ln -s arm-eabi-elfedit arm-none-eabi-elfedit	
+    ln -s arm-eabi-gcc-ar arm-none-eabi-gcc-ar	
+    ln -s arm-eabi-gcov-dump arm-none-eabi-gcov-dump	
+    ln -s arm-eabi-gprof arm-none-eabi-gprof	
+    ln -s arm-eabi-objcopy arm-none-eabi-objcopy	
+    ln -s arm-eabi-size arm-none-eabi-size	
+    ln -s arm-eabi-c++ arm-none-eabi-c++	
+    ln -s arm-eabi-g++ arm-none-eabi-g++	
+    ln -s arm-eabi-gcc-nm arm-none-eabi-gcc-nm	
+    ln -s arm-eabi-gcov-tool arm-none-eabi-gcov-tool	
+    ln -s arm-eabi-ld arm-none-eabi-ld	
+    ln -s arm-eabi-objdump arm-none-eabi-objdump	
+    ln -s arm-eabi-strings arm-none-eabi-strings	
+  fi	
```

Comment:
I think building and configuring a cross compiler is out of the scope of this PKGBUILD. I personally create a separate `makepkg` config, such as `/etc/makepkg-arm64.conf` like so:

```
source /etc/makepkg.conf

CARCH="aarch64"
CHOST="aarch64-linux-gnu"

export ARCH='arm64'
export CROSS_COMPILE='aarch64-linux-gnu-'

CPPFLAGS=""
CFLAGS="-march=armv8-a -O2 -pipe -fstack-protector-strong -fno-plt"
CXXFLAGS="-march=armv8-a -O2 -pipe -fstack-protector-strong -fno-plt"
LDFLAGS=""
```

Then I make a little script to run `makepkg` with this config, such as `/usr/bin/makepkg-arm64`:
```
#!/bin/bash
makepkg --config "/etc/makepkg-arm64.conf" "$@"
```

This way the toolchain can be installed and updated from the Arch repos (`aarch64-linux-gnu-gcc` in this case), and you can cross compile other packages as well.

Reply:
@jakogut - The problem lies with it failing to compile for bare-metal with any other toolchain(See: https://github.com/ARM-software/tf-issues/issues/650) on non-aarch64 hosts. This includes building the same toolchain from source.

Reply:
Apologize, apparently my brain is mush with lack of sleep. | x86_64 hosts cannot compile without that toolchain for initial package creation, as qemu cannot create bare-metal binaries due to a memory mapping error(see above) | i have no problem removing these as it was exclusive for the cross-compiling, and they can be added to a custom makepkg config, as you specified.

Although, id recommend adding notes, as that specific version of the toolchain is the ONLY compatible compiler on x86_64/x86(i686)